src/main/java/org/jabref/model/entry/BibEntry.java

```diff
@@ -112,6 +113,110 @@ public BibEntry(EntryType type) {
         return Optional.empty();
     }
 
+    /**
+     * Map an (empty) field of a BibEntry to a field of a cross-referenced entry.
+     *
+     * @param targetField field name of the BibEntry
+     * @param targetEntry type of the BibEntry
+     * @param sourceEntry type of the cross-referenced BibEntry
+     *
+     * @return the mapped field or null if there is no valid mapping available
+     */
+    private Field getSourceField(Field targetField, EntryType targetEntry, EntryType sourceEntry) {
+        //// 1. Sort out forbidden fields
+        if ((targetField == StandardField.IDS) ||
```

Comment:
you can simply the code a bit:
Create an EnumSet of the fields and check with contains if the field is in the set.
https://docs.oracle.com/javase/8/docs/api/java/util/EnumSet.html

Reply:
Thanks for your suggestion, but I am not sure, if this fits here. First, I have to list the fields anyway, so I don't think, it would simplify the code. Second, the set is accessed only in one single place, so making it a static class member would extend the scope unnecessarily and clutter the class member list. Creating it locally would improve neither code nor performance, I guess. What do you think?